src/main/java/spoon/pattern/PatternBuilder.java

```diff
@@ -120,6 +120,10 @@ public static PatternBuilder create(CtTypeReference<?> templateTypeRef, List<CtE
 		return new PatternBuilder(templateTypeRef, patternModel);
 	}
 
+	public static PatternBuilder create(CtElement patternModel) {
```

Comment:
I expect that most of the patterns will need a reference to a declaring type. So I suggest that all PatternBuilder methods have that reference. It forces clients to think whether they have such a reference and provide it (and it MUST be provided, to assure that matching/generating works well) or to provide null if they are sure that in their special/rare case the pattern is independent on the type.

Note: that there is no way how to add that reference later. It must be added at the beginning of Pattern building process.

But I can live with that new method if you really like it ;-)